logstash-core/lib/logstash/pipeline.rb

```diff
@@ -94,9 +94,9 @@ def initialize(config_str, settings = SETTINGS, namespaced_metric = nil)
     rescue => e
       raise
     end
-    queue = build_queue_from_settings
-    @input_queue_client = queue.write_client
-    @filter_queue_client = queue.read_client
+    @queue = build_queue_from_settings
```

Comment:
Is there any particular reason why an IVAR reference to the queue itself is necessary? I had it in mind that the only access to the queue was via the clients.